File: ios/Runner/pdf/PdfPlatformView.swift
import Flutter
import UIKit

final class PdfPlatformView: NSObject, FlutterPlatformView {
  private let view: PdfRenderView
  private let engine: PdfEngine

  init(engine: PdfEngine) {
    self.engine = engine
    self.view = PdfRenderView(engine: engine)
    super.init()
  }

  func view() -> UIView {
    view
  }
}

File: android/app/src/main/kotlin/com/pixeldev/Docly/pdf/PdfPlatformView.kt
package com.pixeldev.Docly.pdf

import android.content.Context
import android.view.View
import io.flutter.plugin.platform.PlatformView

class PdfPlatformView(
    context: Context,
    private val engine: PdfEngine,
) : PlatformView {
    private val view = PdfRenderView(context, engine)

    override fun getView(): View = view

    override fun dispose() {
        engine.attachInvalidator(null)
    }
}
